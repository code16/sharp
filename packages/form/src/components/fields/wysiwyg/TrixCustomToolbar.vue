<template>
    <div>
        <div class="editor-toolbar">
            <template v-for="part in toolbar">
                <i v-if="part==='|'" class="separator">|</i>
                <button v-else-if="buttons[part]"
                    tabindex="-1"
                    class="fa"
                    :class="buttons[part].icon"
                    v-button-data="buttons[part]"
                >
                </button>
            </template>
        </div>
        <div class="trix-dialogs" data-trix-dialogs>
            <div class="trix-dialog trix-dialog--link" data-trix-dialog="href" data-trix-dialog-attribute="href">
                <div class="trix-dialog__link-fields">
                    <input type="url" name="href" class="trix-input trix-input--dialog" :placeholder="lSub('dialogs.add_link.input_placeholder')" required data-trix-input>
                    <div class="trix-button-group">
                        <input type="button" class="trix-button trix-button--dialog" :value="lSub('dialogs.add_link.link_button')" data-trix-method="setAttribute">
                        <input type="button" class="trix-button trix-button--dialog" :value="lSub('dialogs.add_link.unlink_button')" data-trix-method="removeAttribute">
                    </div>
                </div>
            </div>
        </div>
    </div>
</template>

<script>
    import { buttons } from "./config";
    import { LocalizationBase } from "sharp/mixins";

    export default {
        mixins:[ LocalizationBase('form.wysiwyg') ],
        props: {
            toolbar: Array,
        },
        data() {
            return {
                buttons
            }
        },
        directives: {
            buttonData(el, { value }) {
                let { attribute, action } = value;
                attribute && el.setAttribute('data-trix-attribute', attribute);
                action && el.setAttribute('data-trix-action', action);
            },
        }
    }
</script>