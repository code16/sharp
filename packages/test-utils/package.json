{
    "name": "@sharp/test-utils",
    "private": true,
    "main": "src/index.js",
    "version": "1.0.0"
}