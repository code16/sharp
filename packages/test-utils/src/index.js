export { default as MockInjections } from './MockInjections';
export { default as MockI18n } from './MockI18n';
export { default as QueryComponent } from './QueryComponent';
export { default as MockTransitions } from './MockTransitions';
export { mockSFC, unmockSFC } from './mockSFC';
export { mockProperty, unmockProperty, setter } from './mock-utils';
export { nextRequestFulfilled } from './moxios-utils';

export function wait(delay) {
    return new Promise(resolve => {
        setTimeout(resolve, delay);
    });
}