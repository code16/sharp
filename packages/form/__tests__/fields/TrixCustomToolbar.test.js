import { mount } from '@vue/test-utils';
import Vue from 'vue';
import TrixCustomToolbar from '../../src/components/fields/wysiwyg/TrixCustomToolbar.vue';
import { MockI18n } from "@sharp/test-utils";

describe('trix-custom-toolbar',()=>{
    let wrapper;
    Vue.use(MockI18n, { mockFn: true });

    beforeEach(()=>{
        wrapper = mount(TrixCustomToolbar, {
            propsData : {
                toolbar: ['bold', 'italic', '|', 'link']
            },
            attachToDocument: true
        });
    });

    test('can mount TrixCustomToolbar', ()=>{
        expect(wrapper.html()).toMatchSnapshot();
    });
});