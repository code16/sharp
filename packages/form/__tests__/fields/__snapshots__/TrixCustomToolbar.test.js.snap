// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`trix-custom-toolbar can mount TrixCustomToolbar 1`] = `
<div>
	<div class="editor-toolbar">
		<button tabindex="-1" class="fa fas fa-bold" data-trix-attribute="bold"></button>
		<button tabindex="-1" class="fa fas fa-italic" data-trix-attribute="italic"></button>
		<i class="separator">|</i>
		<button tabindex="-1" class="fa fas fa-link" data-trix-attribute="href" data-trix-action="link"></button>
	</div>
	<div data-trix-dialogs="" class="trix-dialogs">
		<div data-trix-dialog="href" data-trix-dialog-attribute="href" class="trix-dialog trix-dialog--link">
			<div class="trix-dialog__link-fields">
				<input type="url" name="href" placeholder="{{ form.wysiwyg.dialogs.add_link.input_placeholder }}" required="required" data-trix-input="" class="trix-input trix-input--dialog">
				<div class="trix-button-group">
					<input type="button" value="{{ form.wysiwyg.dialogs.add_link.link_button }}" data-trix-method="setAttribute" class="trix-button trix-button--dialog">
					<input type="button" value="{{ form.wysiwyg.dialogs.add_link.unlink_button }}" data-trix-method="removeAttribute" class="trix-button trix-button--dialog">
				</div>
			</div>
		</div>
	</div>
</div>
`;
